### PR TITLE
Test scenarios

### LINKED ISSUE #18929: Permissions: Connecting to/disconnecting from a dapp

### LINKED ISSUE #18940: Decrypt: Encrypt and decrypt message

### LINKED ISSUE #18941: Sign: eth sign, personal sign, signed type data and sign in with ethereum

### PR DESCRIPTION
**Pull request by tmashuang**
## Explanation
Test scenarios for eth sign, personal sign, sign typed data, sign in with ethereum, encrypt & decrypt message, and connecting and disconnecting from dapp

Closes #18941
Closes #18940 
Closes #18929

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

**Comment by benjisclowder** [comment]
Hi @tmashuang , could you please switch the Goerli network recommendation to Sepolia since the former will be deprecated?

File: test/scenarios/11. sign/eth sign.csv [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Steps,Test Steps,Test Data,Expected Result,Notes
+1,Open the extension.,,The Welcome Back screen is shown.,
+2,Proceed to Unlock the wallet.,password (8 characters min).,"The Ether balance is shown on the overview.
+The wallet address is shown on the overview.
+The selected network is Ethereum Mainnet.",
+3,Switch networks to a test network.,e.g. Goerli.,"The Goerli balance is shown on the overview.
+The wallet address is shown on the overview.
+The selected network is Goerli.",
+4,Toggle eth_sign requests advanced setting,,,
```
Comment (PeterYinusa) [inline test/scenarios/11. sign/eth sign.csv:9]: Can we add the steps needed to do this? opening the accounts options menu -> selecting settings -> advanced -> enabling eth_sign.

File: test/scenarios/12. decrypt/encrypt and decrypt.csv [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+The wallet address is shown on the overview.
+The selected network is Ethereum Mainnet.",
+3,Switch networks to a test network.,e.g. Goerli.,"The Goerli balance is shown on the overview.
+The wallet address is shown on the overview.
+The selected network is Goerli.",
+4,Open the test dapp in another tab.,https://metamask.github.io/test-dapp/,,
+5,Proceed to Connect with MetaMask.,,,
+6,Connect with the current account.,,,
+7,Get encryption key.,,The message is in plain text format.,
+8,Encrypt message.,,,
```
Comment (PeterYinusa) [inline test/scenarios/12. decrypt/encrypt and decrypt.csv:13]: Can we add the step for adding a message to encrypt

**Comment by chloeYue** [comment]
> Hi @tmashuang , could you please switch the Goerli network recommendation to Sepolia since the former will be deprecated?

Done. Thanks !

File: test/scenarios/11. sign/eth sign.csv (hunk 1 above)
Comment (chloeYue) [reply test/scenarios/11. sign/eth sign.csv]: Added, thanks!

File: test/scenarios/12. decrypt/encrypt and decrypt.csv (hunk 2 above)
Comment (chloeYue) [reply test/scenarios/12. decrypt/encrypt and decrypt.csv]: Done, thanks!

**Review by PeterYinusa — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]

**Pull request merged**